### ISSUE TITLE
Weighted sampling of train sequences.

### ISSUE THREAD

**Issue by AaronSpieler** [awslabs/gluon-ts#612]
## Description
I propose that we add the feature to do weighted sampling in the InstanceSplitter, and additionally the  option to set a cutoff date before which none will be sampled.
I should not be a breaking change as we can set the default cutoff to None and the default distribution of weights to uniform.

## References
- InstanceSplitter in GluonTS: https://gluon-ts.mxnet.io/api/gluonts/gluonts.transform.html?highlight=instancesplitter#gluonts.transform.InstanceSplitter
- CutOff parameter used in N-BEATS: https://arxiv.org/abs/1905.10437
- Weighted sampling in PyTorch: https://pytorch.org/docs/stable/data.html#torch.utils.data.WeightedRandomSampler

**Comment by mbohlkeschneider** [comment]
+1 on that one!

**Comment by konradsemsch** [comment]
Is there currently any possibility to impact the frequency with which individual time series are sampled/ weighted during model training? This paper: [link](https://arxiv.org/abs/1704.04110) discusses that (more frequent sampling of rare time series that exhibit overall higher volume of transactions), but I'm not sure how to translate that in code at GluonTS current state. Any hints how to go about that?

**Comment by AaronSpieler** [comment]
@konradsemsch 
1) Anything that is sampling related and based on the specific properties of the time series itself is done in the [InstanceSplitter](https://gluon-ts.mxnet.io/api/gluonts/gluonts.transform.html?highlight=instancesplitter#gluonts.transform.InstanceSplitter). So you might find the functionality you looking for there, either by passing certain parameters, e.g. see the `train_sampler` parameter. For example you can control with the specific train sampler how many training examples should be generated from a time series based on its length.
2) Other then that an easy fix could be just preemptively replicating time series in your train set.
3) Regarding weighting the loss im not entirely sure, but I don't think we have that yet.

**Comment by cscheidiger** [comment]
+1 that'd be very helpful!
